website/static/js/fangorn.js

```diff
@@ -426,26 +426,47 @@ function doItemOp(operation, to, from, rename, conflict) {
     }
     orderFolder.call(tb, from.parent());
 
+    var jsonData;
+    if (operation === OPERATIONS.RENAME) {
+        jsonData = JSON.stringify({
+            'action': 'rename',
+            'rename': rename
+        });
+    } else if (operation === OPERATIONS.COPY) {
+        jsonData = JSON.stringify({
+            'action': 'copy',
+            'path': to.data.path || '/',
+            'conflict': conflict,
+            'resource': to.data.resource,
+            'provider': to.data.provider
+        });
+    } else if (operation === OPERATIONS.MOVE) {
+        jsonData = JSON.stringify({
+            'action': 'move',
+            'path': to.data.path || '/',
+            'conflict': conflict,
+            'resource': to.data.resource,
+            'provider': to.data.provider
+        });
+    }
+
     $.ajax({
         type: 'POST',
         beforeSend: $osf.setXHRAuthorization,
-        url: operation === OPERATIONS.COPY ? waterbutler.copyUrl() : waterbutler.moveUrl(),
+        url: waterbutler.buildTreeBeardFileOp(from),
         headers: {
             'Content-Type': 'Application/json'
         },
-        data: JSON.stringify({
-            'rename': rename,
-            'conflict': conflict,
-            'source': waterbutler.toJsonBlob(from),
-            'destination': waterbutler.toJsonBlob(to),
-        })
+        data: jsonData
     }).done(function(resp, _, xhr) {
         if (to.data.provider === from.provider) {
             tb.pendingFileOps.pop();
         }
+    var fromMaterialized = from.data.materialized || '/';
+    var toMaterialized = to.data.materialized;
```

Comment:
If this is only being used once, there's no need to extract it into a variable.  Or if it's being used again, consider putting the alternative initial value (`'/'`) on this line, so you don't have to do it inline below.